### PR TITLE
Adapt FIM json event creation and sending to c++ for whodata and realtime

### PR DESCRIPTION
**Pull request by jotacarma90**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/12103|

## Description
Good team,
this PR is to implement the necessary change in the json creation flow corresponding to the added and modified events of realtime and whodata.
Now all this flow is done in a callbacks in c++, in the future we hope to implement this format for the rest of events.

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
- [x] Source installation
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [x] AddressSanitizer
- Memory tests for Windows
  - [x] Scan-build report

### CODE REVIEW

File: src/syscheckd/src/create_db.c [hunk 1]
```diff
@@ -1180,7 +1184,7 @@ fim_file_data *fim_get_data(const char *file, const directory_t *configuration,
     data->inode = statbuf->st_ino;
     data->dev = statbuf->st_dev;
     data->options = configuration->options;
-    // data->last_event = time(NULL);
+    data->last_event = time(NULL);
```
Comment (FrancoRivero) [inline src/syscheckd/src/create_db.c:1187]: This change can generate problems with PR by Jose Antonio

File: src/syscheckd/src/db/tests/db/ComponentTest/dbInterface/dbTest.h [hunk 2]
```diff
@@ -53,7 +84,7 @@ class DBTestFixture : public testing::Test {
                         300,
                         mockSyncMessage,
                         mockLoggingFunction,
-                        MAX_FILE_LIMIT,
+                        5000,
```
Comment (FrancoRivero) [inline src/syscheckd/src/db/tests/db/ComponentTest/dbInterface/dbTest.h:87]: Could you add a constant for this limits? this is more clear to see!

**Review by FrancoRivero — CHANGES_REQUESTED** [review]
You need add a branch inside wazuh-packages to pass manager's tests

File: src/syscheckd/src/db/tests/db/ComponentTest/dbInterface/dbTest.h (hunk 2 above)
Comment (jotacarma90) [reply src/syscheckd/src/db/tests/db/ComponentTest/dbInterface/dbTest.h:87]: Done

**Review by FrancoRivero — APPROVED** [review]
LGTM

**Review by chemamartinez — APPROVED** [review]

**Pull request merged**